HarmonyOS/Day2 Assignment/JaiJaveria/LoginSignup/entry/src/main/java/com/example/loginsignup/slice/SignupPage.java

```diff
@@ -0,0 +1,57 @@
+package com.example.loginsignup.slice;
+
+import com.example.loginsignup.ResourceTable;
+import ohos.aafwk.ability.AbilitySlice;
+import ohos.aafwk.content.Intent;
+import ohos.agp.components.Button;
+import ohos.agp.components.Component;
+import ohos.agp.components.Text;
+import ohos.agp.components.TextField;
+
+public class SignupPage extends AbilitySlice {
+    @Override
+    protected void onStart(Intent intent) {
+        super.onStart(intent);
+        super.setUIContent(ResourceTable.Layout_signup_page);
+        Button button= (Button) findComponentById(ResourceTable.Id_xSignupSubmit);
+        button.setClickedListener(new Component.ClickedListener() {
+            @Override
+            public void onClick(Component component) {
+                TextField firstName= (TextField) findComponentById(ResourceTable.Id_xFirstName);
+                if(!correctName(firstName.getText()))
+                {
+                    Text text=(Text) findComponentById(ResourceTable.Id_xFirstNameError);
+                    text.setVisibility(Component.VISIBLE);
+                    return;
+                }
+                TextField lastName= (TextField) findComponentById(ResourceTable.Id_xFirstName);
+                if(!correctName(lastName.getText()))
+                {
+                    Text text=(Text) findComponentById(ResourceTable.Id_xLastNameError);
+                    text.setVisibility(Component.VISIBLE);
+                    return;
+                }
+                TextField email= (TextField) findComponentById(ResourceTable.Id_xEmailId);
+                Text text=(Text) findComponentById(ResourceTable.Id_xEmailIdError);
+                if(!correctEmail(email.getText(), text))
+                {
+                    text.setVisibility(Component.VISIBLE);
+                    retua@, 
```

Comment:
Syntax error..!! **Please make sure the succesfully compiled is pushed into the repo**

Reply:
Apologies. Actually it was return; initially and then I started testing it. To type things in the emulator I had to use the virtual keyboard but I sometimes used my laptop's one which edited the code. Since the code was running correctly I directly shifted to adding it on git and thus it reflected here. I will take more care the next time